src/modules/rlm_lua/lua.c

```diff
@@ -782,11 +838,32 @@ int do_lua(rlm_lua_thread_t *thread, REQUEST *request, char const *funcname)
 		goto error;
 	}
 
-	if (lua_pcall(L, 0, 0, 0) != 0) {
+	if (lua_pcall(L, 0, 1, 0) != 0) {
 		char const *msg = lua_tostring(L, -1);
 		REDEBUG("Call to %s failed: %s", funcname, msg ? msg : "unknown error");
 		goto error;
 	}
 
-	return 0;
+	
+	/*
+	 *	the functions should return
+	 */
+	if (lua_isnil(L, -1)) goto error;
+
+	ret = lua_tointeger(L, -1);
+	lua_pop(L, 1);
+
+	/*
+	 *	... and should be a valid "radiusd.*" constant
+	 */
+	ret_str = rlm_lua_constants2str(ret);
+	if (!ret_str) {
+		RDEBUG2("The called %s() returned invalid value", funcname);
```

Comment:
Wrong logging macro.  It should be ``REDEBUG``.  Maybe something like ``Lua function \"%s\" returned invalid rcode \"%s\"`` and then print what was returned.